statsmodels/formula/_manager.py

```diff
@@ -0,0 +1,723 @@
+                )
+            else:
+                _eval_env = eval_env
+                if isinstance(eval_env, patsy.eval.EvalEnvironment):
+                    warnings.warn(EVAL_ENV_WARNING, FutureWarning)
+            if (
+                isinstance(
+                    formula, (patsy.design_info.DesignInfo, patsy.desc.ModelDesc)
+                )
+                or "~" not in formula
+                or formula.strip().startswith("~")
+            ):
+                output = patsy.dmatrix(
+                    formula, data, eval_env=eval_env, return_type=return_type, **kwargs
+                )
+            else:  # "~" in formula:
+                output = patsy.dmatrices(
+                    formula, data, eval_env=eval_env, return_type=return_type, **kwargs
+                )
+            if isinstance(output, tuple):
+                self._spec = output[1].design_info
+            else:
+                self._spec = output.design_info
+            if isinstance(na_action, NAAction):
+                self._missing_mask = getattr(na_action, "missing_mask", None)
+            return output
+
+        else:  # self._engine == "formulaic":
+            import statsmodels.formula
+
+            kwargs = {}
+            if pandas:
+                kwargs["output"] = "pandas"
+            else:
+                kwargs["output"] = "numpy"
+
+            if na_action:
+                kwargs["na_action"] = na_action
+
+            if isinstance(data, (dict, Mapping)):
+                # Work around for no dict support in formulaic
+                if all(np.isscalar(v) for v in data.values()):
+                    # Handle dict of scalars
+                    _data = pd.DataFrame(data, index=[0])
+                else:
+                    _data = pd.DataFrame(data)
+            elif isinstance(data, np.rec.recarray):
+                # workaround no recarray support in formulaic
+                _data = pd.DataFrame.from_records(data)
+            else:
+                _data = data
+
+            _eval_env = eval_env
+            if HAVE_PATSY:
+                from patsy.eval import EvalEnvironment
+
+                if isinstance(eval_env, EvalEnvironment):
+                    warnings.warn(EVAL_ENV_WARNING, FutureWarning)
+
+                    ns = eval_env._namespaces
+                    _eval_env = {}
+                    for val in ns:
+                        _eval_env.update(val)
+            if not isinstance(_eval_env, (int, dict)):
+                raise TypeError("context (eval_env) must be an int or a dict.")
+
+            _ordering = statsmodels.formula.options.ordering
+            if isinstance(formula, self.model_spec_type):
+                _formula = formula
+            if _ordering == "legacy":
+                _formula = self._legacy_orderer(formula, _data, context=_eval_env)
+            else:
+                feature_flags = formulaic.parser.DefaultParserFeatureFlag.TWOSIDED
```

Comment:
I think I might nest these feature flags within the default parser itself. Originally it was intended that these be shared between parsers, but that didn't make a great deal of sense in the end. Also, you can use sets of strings if you want to avoid the import.

Reply:
I prefer to be really explicit.  The import is probably free at this point aside form a line of code as most of formulaic will be loaded already.